dxr/plugins/descriptor/__init__.py

```diff
@@ -9,6 +10,17 @@
 import dxr.indexers
 
 
+def describe_readme(lines):
+    """Return a string that represents a description for the given lines of a
+    presumed readme file or None if it can extract no suitable description.
+    """
+    # For now the heuristic is just the first non-empty line.
+    try:
+        return next(ifilter(None, (line.strip() for line in lines)))
+    except StopIteration:
```

Comment:
Me neither at first. I wrote more_itertools.first() to fill that niche before somebody filled me in.